tree.go

```diff
@@ -54,10 +62,14 @@ type tNumberType struct {
 	scale int
 }
 
-type tSliceType struct {
+type SliceType struct {
```

Comment:
Let's move this definition to `types.go`. Trying to keep `tree.go` to be only unexported AST related definitions and helpers.

Reply:
Should `Definition` and `Field` also be moved?